### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by Martynice**

### CODE REVIEW

File: src/main/java/core/basesyntax/FileWork.java [hunk 1]
```diff
@@ -1,14 +1,46 @@
...
-        return null;
+
+    private static final char LETTER = 'w';
+
+    public String[] readFromFile(String fileName) throws RuntimeException {
+
+        File file = new File(fileName);
+        ArrayList<String> textFromFile = new ArrayList<>();
+        try {
+            Scanner scanner = new Scanner(file);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/FileWork.java:26]: let's use FileReader here

File: src/main/java/core/basesyntax/FileWork.java [hunk 2]
```diff
@@ -1,14 +1,46 @@
...
 public class FileWork {
-    public String[] readFromFile(String fileName) {
-        return null;
+
+    private static final char LETTER = 'w';
+
+    public String[] readFromFile(String fileName) throws RuntimeException {
+
+        File file = new File(fileName);
+        ArrayList<String> textFromFile = new ArrayList<>();
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/FileWork.java:24]: please review your variable names
this list is intended to store words that start with `w`

File: src/main/java/core/basesyntax/FileWork.java [hunk 3]
```diff
@@ -1,14 +1,46 @@
...
+        try {
+            Scanner scanner = new Scanner(file);
+            while (scanner.hasNextLine()) {
+                String[] text = scanner.nextLine().toLowerCase().split(" ");
+                for (String line : text) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/FileWork.java:29]:
```suggestion
                for (String word: line) {
```

File: src/main/java/core/basesyntax/FileWork.java [hunk 4]
```diff
@@ -1,14 +1,46 @@
...
+                    if (line.startsWith(String.valueOf(LETTER))) {
+                        textFromFile.add(line.replaceAll("[\\W]", ""));
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/FileWork.java:31]:
```suggestion
                        textFromFile.add(line.replaceAll("\\W+", ""));
```

File: src/main/java/core/basesyntax/FileWork.java [hunk 5]
```diff
@@ -1,14 +1,46 @@
...
+                    }
+                }
+            }
+            String[] result = new String[textFromFile.size()];
+            for (int i = 0; i < textFromFile.size(); i++) {
+                result[i] = textFromFile.get(i);
+            }
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/FileWork.java:38]: try using `list.toArray(String[] array)` method here

File: src/main/java/core/basesyntax/FileWork.java (hunk 1 above)
Comment (Martynice) [reply src/main/java/core/basesyntax/FileWork.java]: done

File: src/main/java/core/basesyntax/FileWork.java (hunk 2 above)
Comment (Martynice) [reply src/main/java/core/basesyntax/FileWork.java]: renamed

File: src/main/java/core/basesyntax/FileWork.java (hunk 3 above)
Comment (Martynice) [reply src/main/java/core/basesyntax/FileWork.java]: changed

File: src/main/java/core/basesyntax/FileWork.java (hunk 4 above)
Comment (Martynice) [reply src/main/java/core/basesyntax/FileWork.java]: changed

File: src/main/java/core/basesyntax/FileWork.java (hunk 5 above)
Comment (Martynice) [reply src/main/java/core/basesyntax/FileWork.java]: done

**Pull request closed without merging**